### PR TITLE
Add Gov TXs (Change Params, Upgrade Height, DAO(Transfer|Burn)

### PR DESCRIPTION
**Pull request by nodiesBlade**
## Reasoning

This PR is made to allow for progammatic construction for governance-related transactions. For example, this can be used by custodians such as Copper to automate governance transactions, allowing for automation gov tx's (i.e modifying stake weight multiplier and transfers) and enhanced security (by delegation to enterprise/industry grade custodians)

#### govDAOTransfer({ toAddress, amount, action }): MsgProtoGovUpgrade

Adds a MsgProtoGovDAOTransferTxMsg for this transaction.

Returns a `MsgProtoGovDAOTransfer`: The unsigned upgrade message

| Param     | Type        | Description                                                                            |
| --------- | ----------- | -------------------------------------------------------------------------------------- |
| action    | `DAOAction` | the action the dao transfer represent (i.e `transfer` or `burn`)                       |
| amount    | `string`    | the Amount of uPOKT to perform with specified action                                   |
| toAddress | `string`    | the recipient of the dao action once executed for transfers. Not required for burning. |

#### govChangeParam({ paramKey, paramValue, overrideGovParamsWhitelistValidation }): MsgProtoGovUpgrade

Adds a MsgProtoGovChangeParamTxMsg for this transaction.

Returns a `MsgProtoGovChangeParam`: The unsigned upgrade message

| Param                                | Type                 | Description                                                                                     |
| ------------------------------------ | -------------------- | ----------------------------------------------------------------------------------------------- |
| paramKey                             | `GovParameter`, `string`       | the governance parameter key                                                                    |
| paramValue                           | `string`             | the new governance parameter value in ASCII (plain text) format.                                |
| paramValue                           | `string`             | the new governance parameter value in ASCII (plain text) format.                                |
| overrideGovParamsWhitelistValidation | `boolean (optional)` | used to override the validation check for a governance parameter not found in GovParameter enum |

#### govUpgradeHeight({ height, version }): MsgProtoGovUpgrade

Adds a MsgProtoGovUpgradeTxMsg for this transaction.

The GovUpgradeTxMsg is multipurpose. This function acts as an opinionated way of applying the protocol's upgrade height

Returns a `MsgProtoGovUpgrade`: The unsigned upgrade message

| Param   | Type     | Description                              |
| ------- | -------- | ---------------------------------------- |
| height  | `int`    | height to upgrade the protocol's version |
| version | `string` | the new protocol's version               |

#### govUpgradeFeatures({ features }): MsgProtoGovUpgrade

Adds a MsgProtoGovUpgradeTxMsg for this transaction.

The GovUpgradeTxMsg is multipurpose. This function acts as an opinionated way of applying the protocol's upgrade features.

Returns a `MsgProtoGovUpgrade`: The unsigned upgrade message

| Param    | Type       | Description                                                                |
| -------- | ---------- | -------------------------------------------------------------------------- |
| features | `string[]` | array of features `[ {featureName1}:{height1}, {featureName2}:{height2} ]` |

#### govUpgrade({ features, height, version }): MsgProtoGovUpgrade

The GovUpgradeTxMsg is multipurpose, use this only if you are familiar with the tx msg, otherwise rely on `govUpgradeHeight` and `govUpgradeFeatures`

Returns a `MsgProtoGovUpgrade`: The unsigned upgrade message

| Param    | Type       | Description                                                                                                            |
| -------- | ---------- | ---------------------------------------------------------------------------------------------------------------------- |
| height   | `int`      | height to upgrade the protocol's version                                                                               |
| version  | `string`   | the new protocol's version                                                                                             |
| features | `string[]` | array of features to deactivate / activate with the notation: `[ {featureName1}:{height1}, {featureName2}:{height2} ]` |

## How to Test

The changes proposed in this PR can be tested by `pnpm run tests`, additional test cases were created for each new transaction in `transactions.tests.ts`

### CODE REVIEW

**Comment by nodiesBlade** [comment]
To reviewers,

I believe the skeleton of this PR is ready to review with basic validation and test cases for first read. 

Next up is constructing the transaction with the right parameters and ensuring the transaction emitted is accurate to the actual call on my local net, but those changes should be minimal and could be done on a second pass.

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Default ignored files
```
Comment (rem1niscence) [inline .idea/.gitignore]: please remove this file and add the folder to `.gitignore`

File: .idea/modules.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rem1niscence) [inline .idea/modules.xml]: same here, please remove and add to `.gitignore` the folder

File: packages/transaction-builder/src/models/proto/tx-signer.proto [hunk 3]
```diff
@@ -79,3 +79,28 @@ message MsgSend {
...
+    repeated string features = 4;
+}
+
+message MsgUpgrade {
+    bytes address = 1;
+    Upgrade upgrade = 2;
+}
+
+message MsgChangeParam {
+    bytes FromAddress = 1;
```
Comment (rem1niscence) [inline packages/transaction-builder/src/models/proto/tx-signer.proto:103]: I believe that the `FromAddress` field has a typo, the "f" should be lowercase according to the DAO message params https://github.com/pokt-network/pocket-core/blob/897e267924ab1cd59f545515c1ec329324c31ee8/proto/x/gov/gov.proto

File: packages/transaction-builder/src/models/proto/tx-signer.proto [hunk 4]
```diff
@@ -79,3 +79,28 @@ message MsgSend {
     bytes ToAddress = 2;
     string amount = 3;
 }
+
+message MsgDAOTransfer {
+    bytes FromAddress = 1;
+    bytes ToAddress = 2;
```
Comment (rem1niscence) [inline packages/transaction-builder/src/models/proto/tx-signer.proto]: both `FromAddress` and `ToAddress` should be lowercase
```suggestion
    bytes fromAddress = 1;
    bytes toAddress = 2;
```

**Review by rem1niscence — CHANGES_REQUESTED** [review]
peer reviewed with @crisog, changes are mostly on typos and gitignore, after these changes are made let's test this on a localnet if you haven't already.

Also please next time separate linting from new features code

**Comment by nodiesBlade** [comment]
<details>
  <summary>Successful DAO Transfer (Code 0)</summary>

```js
const daoTransfer = tb.govDAOTransfer(
    {
        toAddress: "07a6fca4dea9f01e4c19f301df0d0afac128561b",
        // Amount in uPOKT (1 POKT = 1*10^6 uPOKT)
        action: DAOAction.Transfer,
        amount: "100000",
    }
);
console.log(await tb.submit({txMsg: daoTransfer, fee: "10000"}))
```

Output:
```js
{
  logs: null,
  txHash: 'DA0E9523BD7EA342FFC238C91581C7811E71FA0B4AC3CD4A8BACD8AEDD72766E'
}
```

TX Output:
```js
{
    "hash": "DA0E9523BD7EA342FFC238C91581C7811E71FA0B4AC3CD4A8BACD8AEDD72766E",
    "height": 2,
    "index": 0,
    "proof": {
        "data": null,
        "proof": {
            "aunts": null,
            "index": 0,
            "leaf_hash": null,
            "total": 0
        },
        "root_hash": ""
    },
    "stdTx": {
        "entropy": 7931037553567213,
        "fee": [
            {
                "amount": "10000",
                "denom": "upokt"
            }
        ],
        "memo": "",
        "msg": {
            "type": "gov/msg_dao_transfer",
            "value": {
                "action": "dao_transfer",
                "amount": "100000",
                "from_address": "265e1b97e506019d5de654f2b1b388e52725efbe",
                "to_address": "07a6fca4dea9f01e4c19f301df0d0afac128561b"
            }
        },
        "signature": {
            "pub_key": "dcc7981962d1fe3c2c6689ceefc01f44fdd3b17f00b3a687a0adb944b73cb1cb",
            "signature": "611d9a828c0449f5ca8bfdfdfa77feaf6c1fdb4a04d5f25576df822ecbc2c5444082733d9844ab9f20dc6ea465d0981fb2e68046dd1c3e5c313f6504b290320e"
        }
    },
    "tx": "3AEKWwoVL3guZ292Lk1zZ0RBT1RyYW5zZmVyEkIKFCZeG5flBgGdXeZU8rGziOUnJe++EhQHpvyk3qnwHkwZ8wHfDQr6wShWGxoGMTAwMDAwIgxkYW9fdHJhbnNmZXISDgoFdXBva3QSBTEwMDAwGmQKINzHmBli0f48LGaJzu/AH0T907F/ALOmh6CtuUS3PLHLEkBhHZqCjARJ9cqL/f36d/6vbB/bSgTV8lV234Iuy8LFRECCcz2YRKufINxupGXQmB+y5oBG3Rw+XDE/ZQSykDIOKO3L0qDJp4sO",
    "tx_result": {
        "code": 0,
        "codespace": "",
        "data": null,
        "events": null,
        "info": "",
        "log": "",
        "message_type": "dao_tranfer",
        "recipient": "",
        "signer": "265E1B97E506019D5DE654F2B1B388E52725EFBE"
    }
}
```
</details>


<details>
  <summary>Successful DAO Burn (Code 0)</summary>

```js
const daoTransfer = tb.govDAOTransfer(
    {
        toAddress: "07a6fca4dea9f01e4c19f301df0d0afac128561b",
        action: DAOAction.Burn,
        amount: "10000",
    }
);
console.log(await tb.submit({txMsg: daoTransfer, fee: "10000"}))
```

Output:
```js
{
  logs: null,
  txHash: '849F5BA8C30E4565276341A756A1016782C4C37A6F67FED4929B208540D4245D'
}
```

TX Output:
```js
{
    "hash": "849F5BA8C30E4565276341A756A1016782C4C37A6F67FED4929B208540D4245D",
    "height": 20,
    "index": 0,
    "proof": {
        "data": null,
        "proof": {
            "aunts": null,
            "index": 0,
            "leaf_hash": null,
            "total": 0
        },
        "root_hash": ""
    },
    "stdTx": {
        "entropy": 3023972671233411,
        "fee": [
            {
                "amount": "10000",
                "denom": "upokt"
            }
        ],
        "memo": "",
        "msg": {
            "type": "gov/msg_dao_transfer",
            "value": {
                "action": "dao_burn",
                "amount": "10000",
                "from_address": "265e1b97e506019d5de654f2b1b388e52725efbe",
                "to_address": "07a6fca4dea9f01e4c19f301df0d0afac128561b"
            }
        },
        "signature": {
            "pub_key": "dcc7981962d1fe3c2c6689ceefc01f44fdd3b17f00b3a687a0adb944b73cb1cb",
            "signature": "20648399b637c85e3c1be006efb7887e1970473546bdabee520f2cf362d4f12ab47b7a116716e4f6f105290121c0bf9d6a42f52724be869b5dd787f2725bbc0b"
        }
    },
    "tx": "1wEKVgoVL3guZ292Lk1zZ0RBT1RyYW5zZmVyEj0KFCZeG5flBgGdXeZU8rGziOUnJe++EhQHpvyk3qnwHkwZ8wHfDQr6wShWGxoFMTAwMDAiCGRhb19idXJuEg4KBXVwb2t0EgUxMDAwMBpkCiDcx5gZYtH+PCxmic7vwB9E/dOxfwCzpoegrblEtzyxyxJAIGSDmbY3yF48G+AG77eIfhlwRzVGvavuUg8s82LU8Sq0e3oRZxbk9vEFKQEhwL+dakL1JyS+hptd14fyclu8CyiDu4SHmMmvBQ==",
    "tx_result": {
        "code": 0,
        "codespace": "",
        "data": null,
        "events": null,
        "info": "",
        "log": "",
        "message_type": "dao_tranfer",
        "recipient": "",
        "signer": "265E1B97E506019D5DE654F2B1B388E52725EFBE"
    }
}
```
</details>

<details>
  <summary>Successful DAO Upgrade Version (Code 0)</summary>

```js
const upgradeVersion = tb.govUpgradeVersion(
    {
        upgrade: {
            height: 300,
            version: "0.9.1.4",
        }
    }
);

console.log(await tb.submit({txMsg: upgradeVersion, fee: "10000"}))
```

Output:
```js
{                                                                           
  logs: null,                                                               
  txHash: '8335EB770F3C88503055A18FFDDFEFBDF52C03462C84ED1DE64A37C9614FC2A0'
}        
```

TX Output:
```js
{
    "hash": "8335EB770F3C88503055A18FFDDFEFBDF52C03462C84ED1DE64A37C9614FC2A0",
    "height": 636,
    "index": 0,
    "proof": {
        "data": null,
        "proof": {
            "aunts": null,
            "index": 0,
            "leaf_hash": null,
            "total": 0
        },
        "root_hash": ""
    },
    "stdTx": {
        "entropy": 6500136191087201,
        "fee": [
            {
                "amount": "10000",
                "denom": "upokt"
            }
        ],
        "memo": "",
        "msg": {
            "type": "gov/msg_upgrade",
            "value": {
                "address": "265e1b97e506019d5de654f2b1b388e52725efbe",
                "upgrade": {
                    "Height": "300",
                    "Version": "0.9.1.4"
                }
            }
        },
        "signature": {
            "pub_key": "dcc7981962d1fe3c2c6689ceefc01f44fdd3b17f00b3a687a0adb944b73cb1cb",
            "signature": "68dc05f1ce9b4143454b95f93c175bb18baa25ed6b532ea752e061333ba52d411b7ff794aa3c3b81062bef128fe6b427ea2f37a8bcd0a01987529fbdda335e02"
        }
    },
    "tx": "ugEKOQoRL3guZ292Lk1zZ1VwZ3JhZGUSJAoUJl4bl+UGAZ1d5lTysbOI5Scl774SDAisAhIHMC45LjEuNBIOCgV1cG9rdBIFMTAwMDAaZAog3MeYGWLR/jwsZonO78AfRP3TsX8As6aHoK25RLc8scsSQGjcBfHOm0FDRUuV+TwXW7GLqiXta1Mup1LgYTM7pS1BG3/3lKo8O4EGK+8Sj+a0J+ovN6i80KAZh1KfvdozXgIo4aSlpe76xQs=",
    "tx_result": {
        "code": 0,
        "codespace": "",
        "data": null,
        "events": null,
        "info": "",
        "log": "",
        "message_type": "upgrade",
        "recipient": "",
        "signer": "265E1B97E506019D5DE654F2B1B388E52725EFBE"
    }
}
```
</details>


<details>
  <summary>Successful DAO Upgrade Feature (Code 0)</summary>

```js
const upgradeFeature = tb.govUpgradeFeatures(
    {
        upgrade: {
            features: ["RSCAL:3000"]
        }
    }
);
console.log(await tb.submit({txMsg: upgradeFeature, fee: "10000"}))
```

Output:
```js
{                                                                           
  logs: null,                                                               
  txHash: 'F885B34564A00733EADDE970187D478C98E8C8AB9C6066820BAAA1D99AFC32A9'
}   
```

TX Output:
```js
{
    "hash": "F885B34564A00733EADDE970187D478C98E8C8AB9C6066820BAAA1D99AFC32A9",
    "height": 676,
    "index": 0,
    "proof": {
        "data": null,
        "proof": {
            "aunts": null,
            "index": 0,
            "leaf_hash": null,
            "total": 0
        },
        "root_hash": ""
    },
    "stdTx": {
        "entropy": 635978071284421,
        "fee": [
            {
                "amount": "10000",
                "denom": "upokt"
            }
        ],
        "memo": "",
        "msg": {
            "type": "gov/msg_upgrade",
            "value": {
                "address": "265e1b97e506019d5de654f2b1b388e52725efbe",
                "upgrade": {
                    "Features": [
                        "RSCAL:3000"
                    ],
                    "Height": "1",
                    "Version": "FEATURE"
                }
            }
        },
        "signature": {
            "pub_key": "dcc7981962d1fe3c2c6689ceefc01f44fdd3b17f00b3a687a0adb944b73cb1cb",
            "signature": "058dea56fdd446c26b257c1693062326fd3cfaf5bfc24e5b57102039171649be50020a204704ef102b703b3509f08a914d781bdafb9e75327a95d403343e3d0d"
        }
    },
    "tx": "xQEKRAoRL3guZ292Lk1zZ1VwZ3JhZGUSLwoUJl4bl+UGAZ1d5lTysbOI5Scl774SFwgBEgdGRUFUVVJFIgpSU0NBTDozMDAwEg4KBXVwb2t0EgUxMDAwMBpkCiDcx5gZYtH+PCxmic7vwB9E/dOxfwCzpoegrblEtzyxyxJABY3qVv3URsJrJXwWkwYjJv08+vW/wk5bVxAgORcWSb5QAgogRwTvECtwOzUJ8IqRTXgb2vuedTJ6ldQDND49DSjFtZj4ss2QAQ==",
    "tx_result": {
        "code": 0,
        "codespace": "",
        "data": null,
        "events": null,
        "info": "",
        "log": "",
        "message_type": "upgrade",
        "recipient": "",
        "signer": "265E1B97E506019D5DE654F2B1B388E52725EFBE"
    }
}
```
</details>

<details>
  <summary>Successful DAO Change Params (Code 0)</summary>

```js
const changeParam = tb.govChangeParam(
    {
        paramKey: GovParameter.APPLICATION_MaxApplications,
        paramValue: "50"
    }
);

console.log(await tb.submit({txMsg: changeParam, fee: "10000"}))
```

Output:
```js
{
  logs: null,
  txHash: 'AEB9BFF13403C1CBED916C819BC7BB74B7AFB966960DDC27707FD3F396306316'
}     
```

TX Output:
```js
{
    "hash": "AEB9BFF13403C1CBED916C819BC7BB74B7AFB966960DDC27707FD3F396306316",
    "height": 797,
    "index": 0,
    "proof": {
        "data": null,
        "proof": {
            "aunts": null,
            "index": 0,
            "leaf_hash": null,
            "total": 0
        },
        "root_hash": ""
    },
    "stdTx": {
        "entropy": 6355084816678977,
        "fee": [
            {
                "amount": "10000",
                "denom": "upokt"
            }
        ],
        "memo": "",
        "msg": {
            "type": "gov/msg_change_param",
            "value": {
                "address": "265e1b97e506019d5de654f2b1b388e52725efbe",
                "param_key": "application/MaxApplications",
                "param_value": "IjUwIg=="
            }
        },
        "signature": {
            "pub_key": "dcc7981962d1fe3c2c6689ceefc01f44fdd3b17f00b3a687a0adb944b73cb1cb",
            "signature": "5cc6186f22fef8a2405fbd84e97ce7a5c10f2c898c7557a0449b5974dda6a407d1ee19d15d8a33be56dc1835352225375d1d1cb915ba380753fd0c9292e2d102"
        }
    },
    "tx": "0wEKUgoVL3guZ292Lk1zZ0NoYW5nZVBhcmFtEjkKFCZeG5flBgGdXeZU8rGziOUnJe++EhthcHBsaWNhdGlvbi9NYXhBcHBsaWNhdGlvbnMaBCI1MCISDgoFdXBva3QSBTEwMDAwGmQKINzHmBli0f48LGaJzu/AH0T907F/ALOmh6CtuUS3PLHLEkBcxhhvIv74okBfvYTpfOelwQ8siYx1V6BEm1l03aakB9HuGdFdijO+VtwYNTUiJTddHRy5Fbo4B1P9DJKS4tECKMGAkOin/aQL",
    "tx_result": {
        "code": 0,
        "codespace": "",
        "data": null,
        "events": null,
        "info": "",
        "log": "",
        "message_type": "change_param",
        "recipient": "",
        "signer": "265E1B97E506019D5DE654F2B1B388E52725EFBE"
    }
}
```
</details>


Unit tests:
```sh
@pokt-foundation/pocketjs-transaction-builder:test: PASS tests/transactions.test.ts
@pokt-foundation/pocketjs-transaction-builder:test:   TransactionBuilder Tests
@pokt-foundation/pocketjs-transaction-builder:test:     ✓ Creates a new TransactionBuilder with the provided params (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:     ✓ Can switch the chainID on the fly
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:HappyPaths
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ General happy path: creates properly formatted messages without errors (4 ms)
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:UnhappyPaths:MsgProtoSend
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: same addresses (24 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: amount < 0
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: invalid input for amount
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:UnhappyPaths:MsgProtoAppStake
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: amount to stake too small (below 1 POKT) (5 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: chains empty
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: invalid input for amount (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:UnhappyPaths:MsgProtoNodeStakeTx
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: amount to stake too small (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: chains empty
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: invalid input for amount (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:UnhappyPaths:MsgProtoGovChangeParam
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: paramKey empty (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: paramValue empty
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: invalid governance param without override (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:UnhappyPaths:MsgProtoGovDAOTransfer
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: from address empty
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: from and to the same (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: amount < 0
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: amount not number (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: invalid action
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: action is transfer and toAddress is empty
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:Messages:UnhappyPaths:MsgProtoGovUpgrade
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: from address empty (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: height is zero for gov upgrade raw (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: version is empty for gov upgrade raw
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: height is not 1 for feature gov upgrade raw
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: features was provided for version gov upgrade raw (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: features is empty for feature gov upgrade raw
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: upgrade feature with malformed feature tuple
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Invalid case: upgrade feature with malformed height (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:     TransactionBuilder:createTransaction:HappyPath
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoSend (2 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoAppStake
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoAppUnstake (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoNodeStakeTx (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoNodeUnstake
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoNodeUnjail (1 ms)
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoGovChangeParam
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoGovDAOTransfer
@pokt-foundation/pocketjs-transaction-builder:test:       ✓ Creates a RawTxRequest for MsgProtoGovUpgrade (2 ms)
@pokt-foundation/pocketjs-transaction-builder:test: 
@pokt-foundation/pocketjs-transaction-builder:test: Test Suites: 1 passed, 1 total
@pokt-foundation/pocketjs-transaction-builder:test: Tests:       38 passed, 38 total
@pokt-foundation/pocketjs-transaction-builder:test: Snapshots:   0 total
@pokt-foundation/pocketjs-transaction-builder:test: Time:        1.778 s, estimated 2 s
@pokt-foundation/pocketjs-transaction-builder:test: Ran all test suites.
```

File: packages/transaction-builder/src/models/proto/tx-signer.proto (hunk 3 above)
Comment (nodiesBlade) [reply packages/transaction-builder/src/models/proto/tx-signer.proto:103]: If you take a look at the other protos that are self defined, the previous creator also used upper camel case. I do not know if this has a impact on how the protos are autogenerated, and hence why I followed the same pattern.

For example:
https://github.com/pokt-foundation/pocket-js/blob/02f192873c98345134e6fcb8eabf08ca92a22bf7/packages/transaction-builder/src/models/proto/tx-signer.proto#L78

File: packages/transaction-builder/src/models/proto/tx-signer.proto (hunk 4 above)
Comment (nodiesBlade) [reply packages/transaction-builder/src/models/proto/tx-signer.proto]: Read other comment in reference to this

**Comment by nodiesBlade** [comment]
All good to review, and incremented the version to 2.2.0

I added evidence for localnet testing and unit testing.

File: packages/transaction-builder/src/models/proto/tx-signer.proto (hunk 3 above)
Comment (nodiesBlade) [reply packages/transaction-builder/src/models/proto/tx-signer.proto:103]: ah nevermind, i see what you're saying. Made the protos match - and to answer my own question, the camel case doesn't really matter. Pocket core was just inconsistent in naming them, so I guess we will adopt it here.

File: packages/transaction-builder/src/models/msgs/msg-proto-gov-change-param.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+   * Converts an Msg Object to StdSignDoc
+   * @returns {any} - Msg type key value.
+   * @memberof MsgSend
+   */
+  public toStdTxMsgObj(): any {
+    const data = {
+      FromAddress: Buffer.from(this.fromAddress.toLowerCase(), 'hex'),
+      paramKey: this.paramKey,
+      paramVal: Buffer.from(
```
Comment (rem1niscence) [inline packages/transaction-builder/src/models/msgs/msg-proto-gov-change-param.ts:79]: Is this double `Buffer.from` intentional?

**Comment by nodiesBlade** [comment]
@rem1niscence I noticed you updated the versions a couple of days ago - which are what the merge conflicts are. How would you like me to proceed?

**Review by rem1niscence — DISMISSED** [review]

File: packages/transaction-builder/src/models/msgs/msg-proto-gov-change-param.ts (hunk 5 above)
Comment (nodiesBlade) [reply packages/transaction-builder/src/models/msgs/msg-proto-gov-change-param.ts:79]: Yes, it's intentional. One to base64 and the other is the tsproto expects a buffer of the base64 encoded string

**Comment by nodiesBlade** [comment]
Resolved the merge conflict by upgrading the version to 2.2.0

**Review by rem1niscence — APPROVED** [review]

**Pull request merged**